OracleWebCenterPortal/dockerfiles/12.2.1.4/README.md

```diff
@@ -0,0 +1,308 @@
+
+### 1. Creating containers for WebCenter Portal Server
+
+#### 1.1. Update the environment file 
+
+Create an environment file `webcenter.env.list` file, to define the parameters.
+
+Update the parameters inside `webcenter.env.list` as per your local setup.
+
+```
+#Database Configuration
+DB_DROP_AND_CREATE=<true or false>
+DB_CONNECTION_STRING=<Hostname/ContainerName>:<Database Port>/<DB_PDB>.<DB_DOMAIN>
+DB_RCUPREFIX=<RCU Prefix>
+DB_PASSWORD=<Database Password>
+DB_SCHEMA_PASSWORD=<Schema Password>
+
+#configure container
+ADMIN_SERVER_CONTAINER_NAME=<Admin Server Container Name>
+ADMIN_PORT=<Admin Server Port>
+ADMIN_PASSWORD=<Admin Server Password>
+ADMIN_USERNAME=<Admin Server User Name>
+MANAGED_SERVER_PORT=<Managed Server Port>
+
+# Configure Content Server
+CONFIGURE_UCM_CONNECTION=<true or false>
+#Valid option for socket type are socket,jaxws 
+UCM_SOCKET_TYPE=<UCM Socket Type >
+#Required if UCM_SOCKET_TYPE is jaxws
+UCM_URL=<Configure UCM URL If Socket type is jaxws>
+UCM_HOST=<UCM Host>
+UCM_PORT=<UCM Port>
+UCM_ADMIN_USER=<UCM Admin User>
+
+# Configure Elasticsearch Server
+SEARCH_APP_USERNAME=<Search User Name>
+SEARCH_APP_USER_PASSWORD=<Search User Password>
+```
+
+### 1.2. Admin Container (WCPAdminContainer)
+#### A. Creating and Running Admin Container
+
+Run the following command to create the Admin Server container:
+
+```
+$ docker run -i -t --name $ADMIN_SERVER_CONTAINER_NAME --network=WCPortalNET -p <Any Free Port>:$ADMIN_PORT -v $DATA_VOLUME:/u01/oracle/user_projects --env-file <directory>/webcenter.env.list $WCPortalImageName /bin/bash -c "/u01/oracle/container-scripts/configureOrStartAdminServer.sh; /bin/bash"
+```
+**Note:** Replace variables with values configured in webcenter.env.list
+The above command deletes any previous RCU with the same prefix if **DB_DROP_AND_CREATE=true**
+
+The docker run command creates the container as well as starts the Admin Server in sequence given below:
+
+* Node Manager
+* Admin Server
+
+When the command is run for the first time, we need to create the domain and configure the portal server, so following are done in sequence:
+
+* Loading WebCenter Portal schemas into the database
+* Creating WebCenter Portal domain
+* Configuring Node Manager
+* Starting Node Manager
+* Starting Admin Server
+
+#### B. Stopping Admin Container
+```
+$ docker container stop WCPAdminContainer
+```
+
+#### C. Starting Admin Container
+```
+$ docker container start -i WCPAdminContainer
+```
+
+### 1.3. Portal Container (WCPortalContainer)
+
+#### A. Creating and Running Portal Container
+Run the following command to create the Portal Managed Server container:
+
+```
+$ docker run -i -t --name WCPortalContainer --network=WCPortalNET -p <Any Free Port>:$MANAGED_SERVER_PORT -v $DATA_VOLUME:/u01/oracle/user_projects --env-file <directory>/webcenter.env.list $WCPortalImageName /bin/bash -c "/u01/oracle/container-scripts/configureOrStartWebCenterPortal.sh; /bin/bash"
```

Comment:
Same issue here as above: your container should not be running `/bin/bash`

Reply:
Further, why are you specifying a command on the `run` line anyway? Why doesn't your container image have the correct `ENTRYPOINT` or `CMD` so that this isn't necessary? You can use one of your many environment variables to decide which script to run automatically.